src/components/Articles/ResolutionDates.vue

```diff
@@ -1,90 +1,95 @@
+          <span>Resolution or<br>Court Order Dates</span>
+        </label>
+      </v-col>
+
+      <v-col :cols="isEditMode ? '7' : '9'">
+        <div class="info-text" v-if="isEditMode">
+          Indicate the date of the resolution or court order used to alter the company's share
+          structure or the special rights and restrictions of a class or series of shares:
+        </div>
+        <div class="info-text" v-else>
+          Dates of resolutions or court orders to alter the company's share structure or the
+          special rights or restrictions attached to a class or series of shares:
+        </div>
+      </v-col>
+
+      <v-col cols="2" class="align-right" v-if="isEditMode && !isAdding">
+        <v-btn id="add-resolution-date"
+               class="add-btn"
+               text color="primary"
+               @click="isAdding = true"
+        >
+          <v-icon small>mdi-plus</v-icon>
+          <span>Add</span>
+        </v-btn>
+      </v-col>
+    </v-row>
+
+    <!-- Added Dates -->
+    <v-row no-gutters v-if="haveAddedDates" class="mt-3 mb-1">
+      <v-col cols="3"></v-col>
+      <v-col cols="7">
+        <ul class="resolution-date-list info-text pl-0 mt-2">
+          <li v-for="(date, index) in addedDates"
+              :key="`newResolutionDate-${index}`"
           >
-            <v-icon small>mdi-plus</v-icon>
-            <span>Add</span>
-          </v-btn>
-        </v-col>
-      </v-row>
-
-      <!-- Display NEW Dates -->
-      <v-row no-gutters v-if="addedDates && addedDates.length >= 1" class="my-4">
-        <v-col cols="3"></v-col>
-        <v-col cols="7">
-          <ul class="resolution-date-list info-text pl-0 mt-2">
-            <li v-for="(date, index) in addedDates"
-                :key="`newResolutionDate-${index}`"
-                class="my-n2"
+            <strong class="mr-2">{{date}}</strong>
+            <v-btn v-if="isEditMode"
+                   id="remove-resolution-date"
+                   class="remove-btn mt-n1"
+                   text color="primary"
+                   @click="onRemove(index)"
             >
-              <strong class="mr-2">{{date}}</strong>
-              <v-btn id="remove-resolution-date"
-                     class="remove-btn"
-                     text color="primary"
-                     @click="onRemove(index)"
-              >
-                <v-icon small>mdi-delete</v-icon>
-                <span>Remove</span>
-              </v-btn>
-            </li>
+              <v-icon small>mdi-delete</v-icon>
+              <span>Remove</span>
+            </v-btn>
+          </li>
+        </ul>
+      </v-col>
+    </v-row>
+
+    <!-- Date Picker -->
+    <v-row no-gutters v-if="isAdding" class="mt-4">
+      <v-col cols="3"></v-col>
+      <v-col cols="9" class="mb-n4 pr-3">
+        <date-picker
+          title="Resolution or Court Order Date"
+          nudge-right="80"
+          nudge-top="15"
+          @emitDate="onDate($event)"
```

Comment:
I personally prefer event names more specific like onDateSelected, onDateDeselected etc.

Reply:
I agree.

In this case I tried to match the event handler's name with the emit event name.

Do you think I should change it?

Reply:
fixed

Reply:
👍